google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/project/AppEngineProjectServiceImpl.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.project;
+
+import com.google.cloud.tools.intellij.appengine.cloud.AppEngineEnvironment;
+
+import com.intellij.facet.Facet;
+import com.intellij.facet.FacetManager;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.project.Project;
+import com.intellij.packaging.artifacts.Artifact;
+import com.intellij.packaging.impl.artifacts.ArtifactUtil;
+import com.intellij.psi.xml.XmlFile;
+import com.intellij.psi.xml.XmlTag;
+import com.intellij.remoteServer.configuration.deployment.ArtifactDeploymentSource;
+import com.intellij.remoteServer.configuration.deployment.DeploymentSource;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.idea.maven.project.MavenProject;
+import org.jetbrains.idea.maven.project.MavenProjectsManager;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+
+public class AppEngineProjectServiceImpl extends AppEngineProjectService {
+
+  private AppEngineAssetProvider assetProvider;
+
+  public AppEngineProjectServiceImpl() {
+    assetProvider = AppEngineAssetProvider.getInstance();
+  }
+
+  @Nullable
+  @Override
+  public XmlTag getFlexCompatXmlConfiguration(@NotNull Project project,
```

Comment:
Fair question - this was done for convenience so that the UI code could determine if the flex-compat "type" was 'vm' or 'env'. I think I can refactor this like you suggest.